Use headers only for api and app keys

ssc3 opened the pull request:
Datadog now supports api and app keys in headers. Refactoring ruby lib to use headers only

bkabrda commented on lib/dogapi/common.rb:
```diff
@@ -116,8 +116,10 @@ def request(method, url, extra_params, body, send_json, with_app_key=true)
+          current_url = url + prepare_params(extra_params)
           req = method.new(current_url)
+          req['DD-API-KEY'] = @api_key
+          req['DD-APPLICATION_KEY'] = @application_key if with_app_key
```
The underscore seems to be problematic here - no requests work with it. If I replace it with a hyphen, so `DD-APPLICATION-KEY`, everything seems to work fine.

ssc3 replied:
Done. Thanks!

ssc3 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 # Changes
```
Done

ssc3 commented:
@bkabrda I've incorporated your suggestions and added a unit test to check that api and app keys are actually present. Please do continue your review.

Thanks
